pithos/pithos.py

```diff
@@ -446,6 +465,32 @@ def pandora_ready(*ignore):
 
         self.worker_run('connect', args, pandora_ready, message, 'login')
 
+    def sync_explicit_content_filter_setting(self, *ignore):
+        def fn1(*ignore):
+            pass
+
+        def callback1(*ignore):
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter')
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(False)
+            self.prefs_dlg.explicit_content_filter_checkbutton.set_active(False)
+
+        self.worker_run(fn1, '', callback1)
+
+        if self.pandora.connected:
+            def fn2(*ignore):
+                current_filter_state = self.pandora.explicit_content_filter_state
+                self.filter_state = current_filter_state[0]
+                self.pin_protected = current_filter_state[1]
+
+            def callback2(*ignore):
+                self.prefs_dlg.explicit_content_filter_checkbutton.set_active(self.filter_state)
+                if self.pin_protected:
+                    self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter - PIN Protected')
+                else:
+                    self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(True)
+
+            self.worker_run(fn2, '', callback2)
```

Comment:
Something like this?
```python
    def sync_explicit_content_filter_setting(self, *ignore):
        #reset checkbox to default state
        self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter')
        self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(False)
        self.prefs_dlg.explicit_content_filter_checkbutton.set_active(False)

        if self.pandora.connected:
            def get_filter_and_pin_protected_state(*ignore):
                current_filter_state = self.pandora.explicit_content_filter_state
                self.filter_state = current_filter_state[0]
                pin_protected = current_filter_state[1]
                return pin_protected

            def sync_checkbox(pin_protected):
                self.prefs_dlg.explicit_content_filter_checkbutton.set_active(self.filter_state)
                if pin_protected:
                    self.prefs_dlg.explicit_content_filter_checkbutton.set_label('Explicit Content Filter - PIN Protected')
                else:
                    self.prefs_dlg.explicit_content_filter_checkbutton.set_sensitive(True)

            self.worker_run(get_filter_and_pin_protected_state, (), sync_checkbox)
```
self.filter_state would need to be stored though it's used by on_explicit_content_filter_checkbox later.